### ISSUE TITLE
Implement A Witnet-Based Prediction Market For A Recurring Sporting Event

### ISSUE THREAD

**Issue by Tommytrg** [witnet/truffle-box#15]
# Challenge description

Witnet is a decentralized oracle network (DON) that connects smart contracts to the real, off-chain world. Witnet allows any piece of software to retrieve information published at any web address at any point in time, with complete and verifiable proof of the information's integrity.

The Witnet Protocol achieves this by electing a network of computers which act as a "decentralized oracle" to retrieve, attest and deliver information to smart contracts, with no single point of trust. 

Alongside the [Testnet Incentive Program](https://medium.com/witnet/announcing-the-witnet-testnet-incentive-program-cc7a78b8f274), the Witnet Foundation are offering reward to smart contract and dapp developers that build and test example projects on Witnet. The first of these projects is to build a prediction market for a recurring sporting event. For example, a project to predict the winners and losers of any football match in the Premier League.

# Technical instructions
This bounty is split into two tasks:
- 1. Implement a smart contract in Ethereum which utilizes the Witnet Protocol
- 2. (Optional): Implement a client that consumes the smart contract

#### 1. Smart contract
- Create a Witnet Request that queries the result a sporting event 
- The Witnet Request must contain at least three different sources
- Implement a smart contract using the Witnet Request
- For each event, the smart contract must:
    - Add a prediction option for each sports team (win, loss or draw)
    - Close the prediction market before the sporting event has started
    - Send winning participants a corresponding payout
    - Let a participant know if their prediction was correct or not
    - Let every participant know the result of the event
    
#### 2. Dapp
- Let a user predict the result of the sporting event
- Display the results of all the resolved sporting events
- Allow winners to claim rewards after the event

# Definition of done
 You have successfully completed this bounty when:
 - You have implemented the [Technical instructions](#Technical-instructions)
 - The code is well documented and formatted
 - The code tests core contract features
 - The code has passed a successful Witnet Foundation review, which will check for good practices 

# Submission rules
- Deliver a link to the repository, with documentation of the project
- Deploy the smart contract in göerli testnet
- Deliver the address of the prediction market to the github issue
- Deliver the link to the client

# Compensation
Completed bounty will be rewarded with a total of 500 DAI, in accord with the tasks completed: 
1. 300 DAI for implement the smart contract task.
2. 200 DAI more for implement the dapp

# Resources
- Similar implementation can be found on [Github](https://github.com/stampery-labs)
- Witnet documentation: https://docs.witnet.io/
- Witnet repository: https://github.com/witnet/

# Communication
Join the conversation on the [Witnet Community Discord](https://discor.com/invite/QKEa5gU)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 500.0 DAI (500.0 USD @ $1.0/DAI)  attached to it as part of the witnet fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/witnet/truffle-box/15/4006).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/witnet/truffle-box/15/4006).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $90,891.75 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 264 years, 8 months from now.
Please review their action plans below:


**1) [adivyas99](https://gitcoin.co/adivyas99) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/witnet/truffle-box/15/4006?mutate_worker_action=approve&worker=adivyas99) | [reject worker](https://gitcoin.co/issue/witnet/truffle-box/15/4006?mutate_worker_action=reject&worker=adivyas99))_.**

Will develop a dapp for this problem using ML too!
**2) [montanaflynn](https://gitcoin.co/montanaflynn) has been approved to start work.**

I will research Witnet documentation and then build the smart contract that utilizes the Witnet protocol.

After testing the contract I will create a client dApp that allows users to interact with the contract.

I've built smart contracts similar to this before so I'm sure I can complete the bounty.
**3) [masaun](https://gitcoin.co/masaun) has been approved to start work.**

I will build a dApp of prediction market for a recurring sporting event by using decentralized oracle network (DON) provided by Witnet.
**4) [pengiundev](https://gitcoin.co/pengiundev) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/witnet/truffle-box/15/4006?mutate_worker_action=approve&worker=pengiundev) | [reject worker](https://gitcoin.co/issue/witnet/truffle-box/15/4006?mutate_worker_action=reject&worker=pengiundev))_.**

I'm very interested to develop prediction markets and it'll be a great learning experience for me to do it!
**5) [kifen](https://gitcoin.co/kifen) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/witnet/truffle-box/15/4006?mutate_worker_action=approve&worker=kifen) | [reject worker](https://gitcoin.co/issue/witnet/truffle-box/15/4006?mutate_worker_action=reject&worker=kifen))_.**

I'll create a prediction market smart contract for the Spanish football league (the Laliga) using the Witnet protocol.  Next, I'll write a client (dapp) that consumes the smart contract.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/witnet/truffle-box/15/4006).

**Comment by mlnck** [comment]
@Tommytrg I was really into this one when I first applied .. the whole concept is great. Then I tried the tutorial and it was out of synch with the smart contract dependencies. Spent a handful of hours trying to get things running before I got pulled into other things.

I've got some time again and want to get my dApp finished, so was wondering if the tutorial has been updated to have the app compile and run - if not I'm still going to attempt it! I'll just start from where I left off.
Thanks!
